### PR TITLE
Added Curve25519 signing and validation logic

### PR DESCRIPTION
**Pull request by Tolsi**
## Description

Added Curve25519 signing and validation logic (also known as XEd25519 from [Signal spec](https://signal.org/docs/specifications/xeddsa/)) into trezor-crypto library for Waves blockchain integration #489.

## Testing instructions

There are several tests in the pull request for Waves integration.

## Types of changes

* New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: include/TrustWalletCore/TWCurve.h [hunk 1]
```diff
@@ -15,6 +15,7 @@ enum TWCurve {
     TWCurveSECP256k1              /* "secp256k1" */,
     TWCurveED25519                /* "ed25519" */,
     TWCurveED25519Blake2bNano     /* "ed25519-blake2b-nano" */,
+    TWCurveCurve25519 /* "curve25519" */,
```
Comment (hewigovens) [inline include/TrustWalletCore/TWCurve.h]:
```suggestion
    TWCurveCurve25519             /* "curve25519" */,
```

**Comment by hewigovens** [comment]
Looks good, can you add `sodium` to [trezor-crypto/README.md](https://github.com/TrustWallet/wallet-core/blob/master/trezor-crypto/README.md)?

**Comment by Tolsi** [comment]
> If possible, can you add some tests?

https://github.com/TrustWallet/wallet-core/pull/488/files#diff-841dd9d06d5ab95f92b9a651e051ac4aR4744

**Comment by hewigovens** [comment]
This is included and merged in pr #489, let’s close this now

**Pull request closed without merging**